test/unittest/tests/block/test_block_session_size.cpp

```diff
@@ -0,0 +1,277 @@
+        REQUIRE(cached_sz == sz);
+        validate_and_free_size_block(sz);
+    }
+
+    SECTION("Allocate with two sizes in cache ")
+    {
+        WT_SIZE *sz;
+        WT_SIZE *sz2;
+
+        REQUIRE(__wti_block_size_alloc(session->getWtSessionImpl(), &sz) == 0);
+        REQUIRE(__wti_block_size_alloc(session->getWtSessionImpl(), &sz2) == 0);
+
+        // Construct extent cache with two items.
+        sz->next[0] = sz2;
+        bms->sz_cache = sz;
+        bms->sz_cache_cnt = 2;
+
+        WT_SIZE *cached_sz;
+        REQUIRE(__wti_block_size_alloc(session->getWtSessionImpl(), &cached_sz) == 0);
+        REQUIRE(sz == cached_sz);
+        REQUIRE(sz2 != cached_sz);
+        validate_and_free_size_list(bms, 1);
+        validate_and_free_size_block(cached_sz);
+    }
+}
+
+TEST_CASE("Block session: __wti_block_size_free", "[block_session_size]")
+{
+    std::shared_ptr<MockSession> session = MockSession::buildTestMockSession();
+    WT_BLOCK_MGR_SESSION *bms = session->setupBlockManagerSession();
+
+    /*
+     * FIXME-WT-13451: Update __wti_block_size_free function to test that block is set to null
+     * SECTION("Free with null block manager session -- needs discussion")
+     * {
+     *   std::shared_ptr<MockSession> session_no_bm = MockSession::buildTestMockSession();
+     *   WT_SIZE *sz;
+     *
+     *   REQUIRE(__ut_block_size_alloc(session_no_bm->getWtSessionImpl(), &sz) == 0);
+     *   REQUIRE(sz != nullptr);
+     *
+     *   __wti_block_size_free(session_no_bm->getWtSessionImpl(), sz);
+     *
+     *   REQUIRE(sz == nullptr);
+     * }
+     */
+
+    SECTION("Calling free with cache")
+    {
+        WT_SIZE *sz;
+        REQUIRE(__ut_block_size_alloc(session->getWtSessionImpl(), &sz) == 0);
+
+        __wti_block_size_free(session->getWtSessionImpl(), sz);
+
+        REQUIRE(sz != nullptr);
+        REQUIRE(bms->sz_cache == sz);
+        validate_size_list(bms, 1);
+
+        WT_SIZE *sz2;
+        REQUIRE(__ut_block_size_alloc(session->getWtSessionImpl(), &sz2) == 0);
+        __wti_block_size_free(session->getWtSessionImpl(), sz2);
+
+        REQUIRE(sz != nullptr);
+        REQUIRE(bms->sz_cache == sz2);
+        REQUIRE(bms->sz_cache->next[0] == sz);
+        validate_and_free_size_list(bms, 2);
+    }
+}
+
+TEST_CASE("Block session: __block_size_discard", "[block_session_size]")
+{
+    std::shared_ptr<MockSession> session = MockSession::buildTestMockSession();
+    WT_BLOCK_MGR_SESSION *bms = session->setupBlockManagerSession();
+
+    WT_SIZE *sz, *sz2, *sz3;
+    REQUIRE(__wti_block_size_alloc(session->getWtSessionImpl(), &sz) == 0);
+    REQUIRE(__wti_block_size_alloc(session->getWtSessionImpl(), &sz2) == 0);
+    REQUIRE(__wti_block_size_alloc(session->getWtSessionImpl(), &sz3) == 0);
+
+    // Construct size cache with three items.
+    sz2->next[0] = sz3;
+    sz->next[0] = sz2;
+    bms->sz_cache = sz;
+    bms->sz_cache_cnt = 3;
```

Comment:
A blank line before the `SECTION` would make this slightly easier to read.

Reply:
done